webapi/setaltsig_test.go

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) 2020-2021 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package webapi
+
+import (
+	"context"
+	"crypto/ed25519"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/decred/dcrd/chaincfg/v3"
+	dcrdtypes "github.com/decred/dcrd/rpc/jsonrpc/types/v3"
+	"github.com/decred/slog"
+	"github.com/decred/vspd/database"
+	"github.com/gin-gonic/gin"
+)
+
+const (
+	addrCharset = "123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz"
+	sigCharset  = "0123456789ABCDEFGHJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz+/="
+	hexCharset  = "1234567890abcdef"
+	testDb      = "test.db"
+	backupDb    = "test.db-backup"
+)
+
+var (
+	seededRand           = rand.New(rand.NewSource(time.Now().UnixNano()))
+	feeXPub              = "feexpub"
+	maxVoteChangeRecords = 3
+)
+
+func randBytes(n int) []byte {
+	slice := make([]byte, n)
+	if _, err := seededRand.Read(slice); err != nil {
+		panic(err)
+	}
+	return slice
+}
+
+func TestMain(m *testing.M) {
+	// Set test logger to stdout.
+	backend := slog.NewBackend(os.Stdout)
+	log = backend.Logger("test")
+	log.SetLevel(slog.LevelTrace)
+
+	// Set up some global params.
+	cfg.NetParams = chaincfg.MainNetParams()
+	_, signPrivKey, _ = ed25519.GenerateKey(seededRand)
+
+	// Create a database to use.
+	// Ensure we are starting with a clean environment.
+	os.Remove(testDb)
+	os.Remove(backupDb)
+
+	// Create a new blank database for each sub-test.
```

Comment:
This isn't happening. One database is created, `m.Run()` runs all of the tests, and then the database is torn down once at the end.

Reply:
Changed wording in 3727325